test/fw/ptl/utils/plugins/ptl_report_json.py

```diff
@@ -0,0 +1,155 @@
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+import os
+import sys
+from ptl.utils.pbs_dshutils import DshUtils
+
+
+class PTLJsonData(object):
+    """
+    The intent of the class is to generate json format of PTL test data
+    """
+
+    def __init__(self):
+        self.__du = DshUtils()
+        self.__cmd = [os.path.basename(sys.argv[0])]
+        self.__cmd += sys.argv[1:]
+
+    def get_json(self, data=None, prev_data=None):
+        """
+        Method to generate test data in accordance to json schema
+
+        param data: dictionary of a test case details
+        type data: dict
+
+        returns a formatted dictionary of the data
+        """
+        data_json = None
+        if not prev_data:
+            data_json = {
+                'command': ' '.join(self.__cmd),
+                'user': self.__du.get_current_user(),
+                'product_version': data['pbs_version'],
+                'run_id': int(time.time()),
+                'test_conf': {},
+                'machine_info': data['machinfo'],
+                'testsuites': {},
+                'additional_data': {},
+                'test_summary': {
+                    'result_summary': {
+                        'run': 0,
+                        'succeeded': 0,
+                        'failed': 0,
+                        'errors': 0,
+                        'skipped': 0,
+                        'timedout': 0
+                    },
+                    'test_start_time': str(data['start_time']),
+                    'tests_with_failures': [],
+                    'test_suites_with_failures': []
+                }
+            }
+            if data['testparam']:
+                for param in data['testparam'].split(','):
+                    par = param.split('=', 1)
+                    data_json['test_conf'][par[0]] = par[1]
+        else:
+            data_json = prev_data
+        tsname = data['suite']
+        tcname = data['testcase']
+        if tsname not in data_json['testsuites'].keys():
+            data_json['testsuites'][tsname] = {
+                'module': data['module'],
+                'file': data['file'],
+                'testcases': {}
+            }
+        tsdoc = []
+        if data['suitedoc']:
+            for l in str(data['suitedoc']).strip().split('\n'):
```

Comment:
I'm not sure i understand what you are doing here with tsdoc, it seems like you could just use a regex to replace the tabs and quotes instead of doing it line by line but I might be wrong, can you please explain how and why you are modifying data['suitedoc'] ?

Reply:
For every line in docstring, the line is stripped, and casted as string; the line is then stripped again and tab replaced with space + '\' replaced with '\\' ; followed by appending of this updated line to a list and then the elements of the list are joined with a space separator.

Below is an example: 

Current code:
"docstring": "Smoke test shrink to fit by setting a dedicated time to start in an hour and submit a job that can run for as low as 59 mn and as long as 4 hours. Expect the job''s walltime to be greater or equal than the minimum set.",

As is data['testdoc'] value passed to json file:
"docstring2": "\n        Smoke test shrink to fit by setting a dedicated time to start in an\n        hour and submit a job that can run for as low as 59 mn and as long as\n        4 hours. Expect the job's walltime to be greater or equal than the\n        minimum set.\n        ",

Just tab and new line replaced with space:
"docstring3": "         Smoke test shrink to fit by setting a dedicated time to start in an         hour and submit a job that can run for as low as 59 mn and as long as         4 hours. Expect the job's walltime to be greater or equal than the         minimum set.         ",

Reply:
_"For every line in docstring, the line is stripped, and casted as string; the line is then stripped again"_
you can only strip str objects, so you don't need to cast it to a str after stripping it, which means you don't need to do the stripping a second time either.

_"tab replaced with space + '' replaced with '\' ; followed by appending of this updated line to a list and then the elements of the list are joined with a space separator."_
Again, it just seems like you want to replace a \t and \n with space, which can be done if you just do **re.sub(r"[\t\n]", " ", data['suitedoc'])**, right?
I don't really understand what you are doing with quotes` (replace('\'', '\'\''))`it seems like you are replacing 1 single quote with 2 single quotes? Can you please explain why you are doing that?